pkg/cmd/adm/restart.go

```diff
@@ -3,155 +3,198 @@ package adm
 import (
 	"context"
 	"fmt"
-	"time"
+	"os"
 
 	"github.com/kubesaw/ksctl/pkg/client"
-	"github.com/kubesaw/ksctl/pkg/cmd/flags"
 	"github.com/kubesaw/ksctl/pkg/configuration"
 	clicontext "github.com/kubesaw/ksctl/pkg/context"
 	"github.com/kubesaw/ksctl/pkg/ioutils"
-
 	"github.com/spf13/cobra"
 	appsv1 "k8s.io/api/apps/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/wait"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kubectlrollout "k8s.io/kubectl/pkg/cmd/rollout"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// NewRestartCmd() is a function to restart the whole operator, it relies on the target cluster and fetches the cluster config
+// 1.  If the command is run for host operator, it restart the whole host operator.(it deletes olm based pods(host-operator pods),
+// waits for the new deployment to come up, then uses rollout-restart command for non-olm based - registration-service)
+// 2.  If the command is run for member operator, it restart the whole member operator.(it deletes olm based pods(member-operator pods),
+// waits for the new deployment to come up, then uses rollout-restart command for non-olm based deployments - webhooks)
 func NewRestartCmd() *cobra.Command {
-	var targetCluster string
 	command := &cobra.Command{
-		Use:   "restart -t <cluster-name> <deployment-name>",
-		Short: "Restarts a deployment",
-		Long: `Restarts the deployment with the given name in the operator namespace. 
-If no deployment name is provided, then it lists all existing deployments in the namespace.`,
+		Use:   "restart <cluster-name>",
+		Short: "Restarts an operator",
+		Long: `Restarts the whole operator in the given cluster name. 
+		It restarts the operator and checks the status of the deployment`,
 		Args: cobra.RangeArgs(0, 1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			term := ioutils.NewTerminal(cmd.InOrStdin, cmd.OutOrStdout)
 			ctx := clicontext.NewCommandContext(term, client.DefaultNewClient)
-			return restart(ctx, targetCluster, args...)
+			return restart(ctx, args...)
 		},
 	}
-	command.Flags().StringVarP(&targetCluster, "target-cluster", "t", "", "The target cluster")
-	flags.MustMarkRequired(command, "target-cluster")
 	return command
 }
 
-func restart(ctx *clicontext.CommandContext, clusterName string, deployments ...string) error {
+func restart(ctx *clicontext.CommandContext, clusterNames ...string) error {
+	if clusterNames == nil || len(clusterNames) != 1 {
+		return fmt.Errorf("please provide 1 cluster name to restart the operator e.g `ksctl adm restart host`")
+	}
+	clusterName := clusterNames[0]
```

Comment:
this can be enforced using cobra's `command.Args`

```
  Args: cobra.ExactArgs(1),
```

If we expect the argument to be either `host` or `member`, we can use
```
  Args: cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
  ValidArgs: []string{"host", "member"},
```

Reply:
Thanks for this, updated it